kube/deployment_test.go

```diff
@@ -68,6 +69,140 @@ func TestNewDeployment(t *testing.T) {
+		err.Error())
+}
+
+func TestNewDeploymentHelmConfigured(t *testing.T) {
+	assert := assert.New(t)
+
+	role := deploymentTestLoadRole(assert, "role", "pod-with-valid-pod-anti-affinity.yml")
+	if role == nil {
+		return
+	}
+
+	settings := ExportSettings{
+		CreateHelmChart: true,
+		Repository:      "the_repos",
+	}
+
+	grapher := FakeGrapher{}
+
+	deployment, svc, err := NewDeployment(role, settings, grapher)
+
+	assert.NoError(err)
+	assert.Nil(svc)
+	assert.NotNil(deployment)
+	assert.Equal(deployment.Get("kind").String(), "Deployment")
+	assert.Equal(deployment.Get("metadata", "name").String(), "role")
+
+	config := map[string]interface{}{
+		"Values.sizing.role.count":                 "1",
+		"Values.sizing.role.affinity.nodeAffinity": "snafu",
+		"Values.kube.registry.hostname":            "docker.suse.fake",
+		"Values.kube.organization":                 "splat",
+		"Values.env.KUBE_SERVICE_DOMAIN_SUFFIX":    "domestic",
+	}
+
+	deploymentYAML, err := testhelpers.RenderNode(deployment, config)
+	if !assert.NoError(err) {
+		return
+	}
+
+	expectedYAML := `---
+# The role role contains the following jobs:
+
+apiVersion: "extensions/v1beta1"
+kind: "Deployment"
+metadata:
+  name: "role"
+  labels:
+    skiff-role-name: "role"
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      skiff-role-name: "role"
+  template:
+    metadata:
+      name: "role"
+      labels:
+        skiff-role-name: "role"
+      annotations:
+        checksum/config: 08c80ed11902eefef09739d41c91408238bb8b5e7be7cc1e5db933b7c8de65c3
+    spec:
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                - key: "skiff-role-name"
+                  operator: "In"
+                  values:
+                  - "role"
+              topologyKey: "beta.kubernetes.io/os"
+            weight: 100
+        nodeAffinity: "snafu"
+      containers:
+      - env:
+        - name: "KUBERNETES_NAMESPACE"
+          valueFrom:
+            fieldRef:
+              fieldPath: "metadata.namespace"
+        - name: "KUBE_SERVICE_DOMAIN_SUFFIX"
+          value: "domestic"
+        image: "docker.suse.fake/splat/the_repos-role:bfff10016c4e9e46c9541d35e6bf52054c54e96a"
+        lifecycle:
+          preStop:
+            exec:
+              command:
+              - "/opt/fissile/pre-stop.sh"
+        livenessProbe: ~
+        name: "role"
+        ports: ~
+        readinessProbe: ~
+        resources: ~
+        securityContext: ~
+        volumeMounts: ~
+      dnsPolicy: "ClusterFirst"
+      imagePullSecrets:
+      - name: "registry-credentials"
+      restartPolicy: "Always"
+      terminationGracePeriodSeconds: 600
+      volumes: ~
+`
+	assert.Equal(expectedYAML, string(deploymentYAML))
```

Comment:
I disliked it because of `subset` functionality. I.e. when it is not detected when fissile creates more than expected. I might look into a making a `IsYAMLEqual`.

Reply:
Done with commit abbd63a3a9fbf53076745a0609423858731db857